### ISSUE TITLE
Wagtail 2.9/Django Sites Framework

### ISSUE THREAD

**Issue by sekanitembo** [rkhleics/wagtailmenus#366]
Are there plans to update wagtailmenus to be used with wagtail 2.9? I'm running into issues with the current version and the django sites framework's middleware setting `request.site`. Specifically, it looks like this function in `wagtailmenus/utils/misc.py` is causing the issue:
```
def get_site_from_request(request, fallback_to_default=True):
    if getattr(request, 'site', None):
        return request.site
    if fallback_to_default:
        return Site.objects.filter(is_default_site=True).first()
    return None
```

With `'wagtail.core.middleware.SiteMiddleware'` removed from MIDDLEWARE and `'django.contrib.sites.middleware.CurrentSiteMiddleware'` installed, I receive the following error:
```
Traceback (most recent call last):
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/core/handlers/base.py", line 145, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/core/handlers/base.py", line 143, in _get_response
    response = response.render()
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/template/response.py", line 105, in render
    self.content = self.rendered_content
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/template/response.py", line 83, in rendered_content
    return template.render(context, self._request)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/test/utils.py", line 95, in instrumented_test_render
    return self.nodelist.render(context)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/template/base.py", line 936, in render
    bit = node.render_annotated(context)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/template/base.py", line 903, in render_annotated
    return self.render(context)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/test/utils.py", line 95, in instrumented_test_render
    return self.nodelist.render(context)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/template/base.py", line 936, in render
    bit = node.render_annotated(context)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/template/base.py", line 903, in render_annotated
    return self.render(context)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/templatetags/cache.py", line 47, in render
    value = self.nodelist.render(context)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/template/base.py", line 936, in render
    bit = node.render_annotated(context)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/template/base.py", line 903, in render_annotated
    return self.render(context)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/template/library.py", line 192, in render
    output = self.func(*resolved_args, **resolved_kwargs)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/wagtailmenus/templatetags/menu_tags.py", line 31, in main_menu
    return menu_class.render_from_tag(
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/wagtailmenus/models/menus.py", line 1060, in render_from_tag
    return super().render_from_tag(
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/wagtailmenus/models/menus.py", line 89, in render_from_tag
    instance = cls._get_render_prepared_object(
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/wagtailmenus/models/menus.py", line 114, in _get_render_prepared_object
    instance = cls.get_from_collected_values(ctx_vals, opt_vals)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/wagtailmenus/models/menus.py", line 1076, in get_from_collected_values
    return cls.get_for_site(contextual_vals.current_site)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/wagtailmenus/models/menus.py", line 1083, in get_for_site
    instance, created = cls.objects.get_or_create(site=site)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/db/models/query.py", line 559, in get_or_create
    return self.get(**kwargs), False
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/db/models/query.py", line 404, in get
    clone = self._chain() if self.query.combinator else self.filter(*args, **kwargs)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/db/models/query.py", line 904, in filter
    return self._filter_or_exclude(False, *args, **kwargs)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/db/models/query.py", line 923, in _filter_or_exclude
    clone.query.add_q(Q(*args, **kwargs))
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1351, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1378, in _add_q
    child_clause, needed_inner = self.build_filter(
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1285, in build_filter
    self.check_related_objects(join_info.final_field, value, join_info.opts)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1123, in check_related_objects
    self.check_query_object_type(value, opts, field)
  File "/home/<username>/.local/share/virtualenvs/<project venv>/home/<username>/.pyenv/shims/python/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1104, in check_query_object_type
    raise ValueError(
ValueError: Cannot query "example.com": Must be "Site" instance.
```

According to the [release notes](https://docs.wagtail.io/en/stable/releases/2.9.html#sitemiddleware-and-request-site-deprecated) `request.site` should be replaced with `Site.find_for_request(request)` in Python and `{% wagtail_site %}` in templates.  It looks like the deprecated approach is used in a couple of other places:
- `wagtailmenus/tests/templates/sub_menu-tag-used-directly.html`
- `wagtailmenus/utils/tests/test_misc.py`

I could try to replace the usages of `get_site_from_request`/`request.site` myself (in a fork) if that would be helpful.

Thanks!

**Comment by ababic** [comment]
Hi @sekanitembo. Yes, the intention is to support Wagtail 2.9 - just struggling to find any time to do the update myself. If you are able to help out, that would be super awesome. I'll happily review the PR for you.

**Comment by sekanitembo** [comment]
Hey @ababic. I have a local branch setup with what I think will work for Wagtail 2.9 support, but I have questions about the tox/travis testing. It looks like the ``dj111: Django>=1.11,<2.1`` line in tox.ini is skipping Django 1.11 testing. I'm currently messing with lowering the upper range to ``<2.0`` to test my update against Django 1.11 since the docs still seem to imply that it's supported, but I'm having difficulties. 

My main questions are: should I worry about testing that? Is still supported? 

I noticed a lot of the other tox tests errored or failed against other envs, also (with the master branch), but the envs specified for travis seem to work ok. Is that the main threshold for release readiness? Is there a path/steps I'm missing to get all those tox tests passing?

Sorry for all the questions, and I can provide more details if you want, but just wanted to know how thorough the project is with supporting across all those versions in the docs.

**Comment by sekanitembo** [comment]
I finally got sqlite3 downgraded for Python 3.4 and that seems to have taken care of the issue I was facing with running the py34-dj111-wt2 tox env test. I'll see if I can't fix the other ones I've seen failing, too, and get the PR ready.

**Comment by ababic** [comment]
Hi @sekanitembo. Really appreciate you taking this on. If you're struggling with environment setup locally, I've found it's really useful to setup Travis CI to run on your fork - that way, there's far less setup to do locally, and builds are more likely to be consistent. Wagtailmenus is open source, so you shouldn't have to pay anything. Running tests in the cloud is far more convenient, as you can do other things while you're waiting - You can also share links to the failures (which will allow me to better help out if you have more problems).

**Comment by ababic** [comment]
Resolved in eab6762
